src/App.js

```diff
@@ -1,22 +1,55 @@
-import logo from './logo.svg';
 import './App.css';
+import tasks from "./data/tasks.json"
+import GenericTable from './components/table';
+import { useState } from 'react';
+import GenericModal from './components/modal/modal';
+import AddForm from './components/form/addForm';
+import EditForm from './components/form/editForm';
+import DeleteForm from './components/form/deleteForm';
 
 function App() {
+  const [data, setData] = useState(tasks);
+  const [task, setTask] = useState({});
+
+  const tableHeading = ["name", "description", "dueDate", "status"];
+
+  const addFunc = () => {
+    Object.keys(task).length === 4 && 
+    <>
+      {setData([
+        ...data,
+        task
+      ])}
+      {setTask({})}
+    </>
+      
+  }
+
+  const editFunc = () => {
+    Object.keys(task).length === 4 && 
```

Comment:
why to use number?

Reply:
I have used length === 4 in order to check if there are 4 values in the object. There is no need for that in the Edit function so I have removed it from there but I have included this in Add function to avoid adding an empty object or an object with less that 4 values.